progress/README.md

```diff
@@ -0,0 +1,313 @@
+  ```json
+  {
+    "message": "Progress document retrieved successfully.",
+    "count": 1,
+    "data": [
+      {
+        "date": 1683590400000,
+        "createdAt": 1683650576402,
+        "blockers": "None",
+        "completed": "Monitoring for issues and performance",
+        "planned": "Plan for future enhancements",
+        "type": "user",
+        "userId": "NLFSj7Kz30oHgolfIZtJ"
+      }
+    ]
+  }
+  ```
+  GET /progresses?taskId=d757s743d3c54bdf61a6
+  ```json
+  {
+    "message": "Progress document retrieved successfully.",
+    "count": 1,
+    "data": [
+      {
+        "date": 1683676800000,
+        "createdAt": 1683690250824,
+        "blockers": "None",
+        "completed": "Monitoring for issues and performance",
+        "planned": "Plan for future enhancements",
+        "type": "task",
+        "userId": "SooJK37gzjIZfFNH0tlL",
+        "taskId": "d757s743d3c54bdf61a6"
+      }
+    ]
+  }
+  ```
+
+## GET /progresses/range
+
+Retrieves the progress records for a particular user or task within the specified date range, from start to end date.
+
+- **Params**
+  None
+- **Query**
+  Note : Only one in userId or taskId parameter is allowed per request
+
+  - userId : Specifies the ID of the User whose progress we are interested in
+  - taskId : Specifies the ID of the Task whose progress we are interested in
+  - startDate : Specifies the start date of the date range in ISO 8601 format (e.g. "2023-05-15"). Only progress data on or after this date will be included in the results.
+  - endDate : Specifies the end date of the date range in ISO 8601 format (e.g. "2023-05-24"). Only progress data on or before this date will be included in the results.
+    Note: The date should be in the ISO 8601 format i.e YYYY-MM-DD
+- **Body**  
+  None
+- **Headers**  
+  None
+- **Cookie**  
+  None
+- **Success Response:**
+
+  - **Code:** 200
+
+    - **Content:**
+
+    ```json
+    {
+      "message": "Progress document retrieved successfully.",
+      "data": {
+        "startDate": "YYYY-MM-DD",
+        "endDate": "YYYY-MM-DD",
+        "progressRecords": {
+          "YYYY-MM-DD": "boolean",
+          "YYYY-MM-DD": "boolean",
+          "YYYY-MM-DD": "boolean"
+        }
+      }
+    }
+    ```
+
+- **Example:**
+  GET /progresses/range?userId=SooJK37gzjIZfFNH0tlL&startDate=2023-05-01&endDate=2023-05-10
+    Note: The date passe in the query params is of ISO 8601 format i.e YYYY-MM-DD
+  ```json
+  {
+    "message": "Progress document retrieved successfully.",
+    "data": {
+      "startDate": "2023-05-01",
+      "endDate": "2023-05-10",
+      "progressRecords": {
+        "2023-05-01": true,
+        "2023-05-02": true,
+        "2023-05-03": true,
+        "2023-05-04": true,
+        "2023-05-05": true,
+        "2023-05-06": true,
+        "2023-05-07": true,
+        "2023-05-08": false,
+        "2023-05-09": true,
+        "2023-05-10": true
+      }
+    }
+  }
+  ```
```

Comment:
please add a 404 and internal server error response

Reply:
Done.